src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -663,7 +688,7 @@ internal Region CreateRegion(PhysicalLocationVersionOne v1AnnotationLocation, Ph
             // So only copy the v1 annotations that refer to the same file as the location.
             if (v1PhysicalLocation != null && v1AnnotationLocation.Uri == v1PhysicalLocation.Uri)
             {
-                region = CreateRegion(v1PhysicalLocation.Region);
+                region = CreateRegion(v1PhysicalLocation);
```

Comment:
As above, your other option is to pass `v1PhysicalLocation.Uri` as a second argument.